### ISSUE TITLE
SpringBatchTest does not work ootb with SpringBoot

### ISSUE THREAD

**Issue by eiswind** [spring-projects/spring-batch#4233]
Given the following job definition with Spring Boot 3-RC2:

```java
@Configuration
public class HelloBatch {

	public static void main(String[] args) {
		SpringApplication.run(BatchApplication.class, args);
	}
	@Bean
	public Step step(JobRepository jobRepository, PlatformTransactionManager tx) {
		return new StepBuilder( "step1", jobRepository)
				.tasklet((contribution, chunkContext) -> {
					System.out.println("Hello, World!");
					return RepeatStatus.FINISHED;
				}, tx).build();
	}

	@Bean
	public Job job(JobRepository jobRepository, Step step) {
		return new JobBuilder("job", jobRepository)
				.start(step)
				.build();
	}
}
```

The following test (as suggested in the javadoc) does fail:

```java
@SpringBatchTest
@SpringJUnitConfig(HelloBatch.class)
public class HelloBatchTest {

	@Autowired
	private JobLauncherTestUtils jobLauncherTestUtils;

	@Test
	public void testHello(@Autowired Job job) throws Exception {

		jobLauncherTestUtils.setJob(job);

		var jobExecution =
				jobLauncherTestUtils.launchJob();

		assertEquals(BatchStatus.COMPLETED, jobExecution.getStatus());

	}
}
```

the exception goes here
```
Error creating bean with name 'jobLauncherTestUtils': Unsatisfied dependency expressed through method 'setJobRepository' parameter 0:
```

I could workaround by making the test a SpringBootTest and disabling the auto-configured joblauncher but i guess that is not the intented solution.

**Comment by fmbenhassine** [comment]
Thank you for opening this ticket. This is a valid issue, and is related to the new way test utilities are configured with `@SpringBatchTest` in v5. Test utilities should _not_ autowire dependencies. I noticed that this is inconsistent as of `5.0.0-RC2` (the job under test is not autowired while the job repository and job launcher are). This has been fixed in 05bbd3f7ea8b909942629e5bdaae94dfdfa2a7f1.

Now on the Boot side, things are a bit different with regard to the setup of the test context. I noticed that without importing the `HelloBatch.class` and using `@SpringBootTest` the test you shared passes. This is actually how I am expecting things to work in a Boot world: the simple fact of using `@SpringBootTest` should import my job configuration (without having to do it manually). This is not a Javadoc issue because the sample in the Javadoc of Spring Batch works with Spring Batch alone.

Could you please give the latest snapshots a try? I am expecting the following test to pass with the latest snapshots:

```java
import org.junit.jupiter.api.Test;

import org.springframework.batch.core.BatchStatus;
import org.springframework.batch.test.JobLauncherTestUtils;
import org.springframework.batch.test.context.SpringBatchTest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import static org.junit.jupiter.api.Assertions.assertEquals;

@SpringBatchTest
@SpringBootTest
public class HelloBatchTest {
    
    @Autowired
    private JobLauncherTestUtils jobLauncherTestUtils;
    
    @Test
    public void testHello() throws Exception {
        var jobExecution =
                jobLauncherTestUtils.launchJob();

        assertEquals(BatchStatus.COMPLETED, jobExecution.getStatus());
    }

}
```

**Comment by eiswind** [comment]
I can confirm that the test passes, but it runs the job twice. (The JobLauncher from the AutoConfiguration...)

I added
@SpringBootTest
@SpringBatchTest
@TestPropertySource(properties = "spring.batch.job.enabled=false")
public class HelloBatchTest {

then everything is fine.

Maybe this should be the default in a test?
